viv_utils/emulator_drivers.py

```diff
@@ -160,6 +175,31 @@ def add_hook(self, hook):
     def remove_hook(self, hook):
         self._hooks.remove(hook)
 
+    def remove_default_viv_hooks(self):
+        """
+        vivisect comes with default emulation hooks (imphooks) that emulate
+         - API calls, e.g. GetProcAddress
+         - abstractions of library code functionality, e.g. _alloca_probe
+
+        in our testing there are inconsistencies in the hook implementation, e.g. around function returns
+        this function removes all imphooks that do not emulate a return
+        """
+        for name in (
+            "ntdll.seh3_prolog",
+            "ntdll.seh4_prolog",
+            "ntdll.seh4_gs_prolog",
+            "ntdll.seh3_epilog",
+            "ntdll.seh4_epilog",
+            "ntdll.eh_prolog",
+            "ntdll._alloca_probe",
+            "ntdll.gs_prolog",
+        ):
```

Comment:
there's still a bit of magic around these, but I think it's user-friendlier than letting the consumer specify which APIs to remove, alternatively, we could
1. remove all default hooks
2. only allow specified hooks (again, users would need to figure these out)

Reply:
i'd prefer to have an allowlist. i'm concerned that viv may add new hooks in future versions and break things here and it will be tricky to track down. using an explicit allowlist ensures that we've reviewed each hook and approve it.

Reply:
i'd also tend to think the allowlist may operate on a per-application level, rather than within viv utils? unless we keep documentation within viv-utils on what we've found.

therefore, i'd recommend:
  - add helper at top level here (i dont think it has to be on the mixin class): `remove_default_viv_hooks`
  - add helper at top level: `add_default_viv_hook_by_name`?
  - put configuration (allowlist) within floss/capa?

Reply:
yes, perfect, that makes a lot of sense, thanks for the details!

Reply:
how are the changes in 63a9aed?

Reply:
example usage in tests: https://github.com/williballenthin/viv-utils/pull/35/files#diff-0d9318c26b3318c879053d68b31196c4702cc501e1929af3322126965bb26e88R256